app/models/manageiq/providers/openstack/cloud_manager/cloud_volume.rb

```diff
@@ -56,8 +56,15 @@ def raw_update_volume(options)
   def validate_delete_volume
     msg = validate_volume
     return {:available => msg[:available], :message => msg[:message]} unless msg[:available]
-    if with_provider_object(&:status) == "in-use"
-      return validation_failed("Create Volume", "Can't delete volume that is in use.")
+    begin
+      if with_provider_object(&:status) == "in-use"
+        return validation_failed("Delete Volume", "Can't delete volume that is in use.")
+      end
+    # NoMethodError here usually means the above block got a nil while checking the volume
+    # status, probably because the user already deleted the volume. Assume that's the case
+    # and show a reassuring message.
+    rescue NoMethodError
```

Comment:
Hi @mansam , thanks for the PR!

I think there can be other cases when NoMethodError is raised than ```Volume already queued for deletion```.

We can merge this in case of emergency, but I'd like discuss changing the line ```with_provider_object(&:status) == "in-use"``` to stop connecting to OpenStack and reading the ```status``` information from MIQ database record instead. Do you know if this would be possible? Thanks!

Reply:
Probably possible, I'll try it and see if I run into any problems.